be/src/exec/exec_node.cpp

```diff
@@ -636,12 +637,14 @@ Status ExecNode::eval_conjuncts(const std::vector<ExprContext*>& ctxs, vectorize
             continue;
         } else if (0 == true_count) {
             // all not hit, return
+            filter->assign(chunk->num_rows(), 0);
             chunk->set_num_rows(0);
             return Status::OK();
         } else {
             bool all_zero = false;
             vectorized::ColumnHelper::merge_two_filters(column, raw_filter, &all_zero);
             if (all_zero) {
+                filter->assign(chunk->num_rows(), 0);
```

Comment:
as above

Reply:
All right, I could move the filter assign operation to `NLJoinOperator`, to avoid affect other procedures.